### ISSUE TITLE
Trendnet TV-IP321PI Flash works but camera doesn't come alive

### ISSUE THREAD

**Issue by NormHenderson** [scottlamb/hikvision-tftpd#7]
Hi, If I should be posting this somewhere else, apologies, let me know where...

I have several of these cameras and have had varying success upgrading via the normal GUI, from firmware 5.1.x (not sure which, maybe 5.1.2) to 5.3.3 and then 5.3.4. This tool was a huge success in recovering one cam that failed the upgrade to 5.3.3. SO first - thank you so much!

Flush with success I then tried to upgrade a new out-of-box unit to 5.3.3 using this tool, without first trying the GUI. The flash appeared to succeed ("done" message) however, after power cycling the camera it remains on the 192.0.0.64 address periodically sending out the magic number. I have tried repeating the flash without success; using the reset button also didn't change the behavior. What next? The older firmware versions are not published on the Trendnet site anymore.

**Comment by scottlamb** [comment]
Sorry for the very delayed response. I haven't gone through issues here in a while. Did you find an answer to your question? You might have better luck on the ipcamtalk.com forums; there are a lot of knowledgeable folks here. But since you asked here...

It's on the 192.0.0.64 address; does it respond to pings? did you try accessing the web interface there?

**Comment by NormHenderson** [comment]
Hi Scott, thanks very much, your reply was timely since I just received the serial connector. No it doesn't respond to pings nor can I access the web interface. Here is what the serial console shows, how to fix?
U-Boot 2010.06-90523 (Oct 17 2014 - 17:35:23)

DRAM:  128 MiB
Flash: 16 MiB
Hit Ctrl+u to stop autoboot:  0
UP_PORT : phy status change : LINK=DOWN : DUPLEX=FULL : SPEED=100M
UP_PORT : phy status change : LINK=UP : DUPLEX=FULL : SPEED=100M
|NUL ethaddr| TFTP server not found
load kernel to 0x80007fc0 ... Done!
## Booting kernel from Legacy Image at 80007fc0 ...
   Image Name:   Linux-3.0.8
   Image Type:   ARM Linux Kernel Image (uncompressed)
   Data Size:    3481816 Bytes = 3.3 MiB
   Load Address: 80008000
   Entry Point:  80008000
   XIP Kernel Image ... OK
OK

Starting kernel ...
øStarting udev:      [ OK ]
mount: mounting /dev/mtdblock5 on /mnt failed: Invalid argument
APP partition is broken!
The system is going down NOW!
Sent SIGTERM to all processes
Sent SIGKILL to all processes
Requesting system reboot

Thanks for any assistance, Norm

**Comment by NormHenderson** [comment]
Actually - with the serial console still connected (but without actually doing anything!) and using hikvision_tftpd.py once again, I was able to successfully flash firmware 5.3.3 and then 5.3.4 and the camera is now working normally. Case closed!

For any others, here is the console setup for Putty serial mode: 115200 baud, 8 data bits 1 stop bit, no parity. no flow control. The connector pinout is, from L to R (nearest the reset button): Common - Rx data (from camera) - Tx data (to camera) - 3.3V (unused). The connector was quite difficult to insert but it worked. I used USB to serial adapter FTDI232 set to 3.3V (not 5v).
